configure screenshare audio track to appear in transcriptions and room I/O

synthobotics opened the pull request:
fix: support screenshare audio in transcription and room I/O

- Update find_micro_track_id() to detect both microphone and screenshare audio tracks
- Update _ParticipantLegacyTranscriptionOutput to handle screenshare audio sources
- Fixes issue where screenshare audio was not being processed by STT nodes

Closes #2400

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/livekit/agents?pullRequest=3098) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/livekit/agents?pullRequest=3098) before we can accept your contribution.<br/><hr/>**sri** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/livekit/agents?pullRequest=3098) it.</sub>

longcw commented on livekit-agents/livekit/agents/voice/room_io/_output.py:
```diff
@@ -242,7 +242,7 @@ def _on_track_published(
         if (
             not self._is_local_proxy_participant(participant)
-            or track.source != rtc.TrackSource.SOURCE_MICROPHONE
+            or track.source not in [rtc.TrackSource.SOURCE_MICROPHONE, rtc.TrackSource.SOURCE_SCREENSHARE_AUDIO]
```
I think it's better to make it configurable and by default we listen to microphone since most of the users won't need to listen to the audio from screen share.

I made a similar pr before https://github.com/livekit/agents/pull/2719 but closed since I don't have a good way to test it. Could you add an option to the `RoomInputOption`?

longcw replied:
The output is the track published by agent, it should always be SOURCE_MICROPHONE?

longcw commented on livekit-agents/livekit/agents/voice/room_io/_output.py:
```diff
@@ -398,7 +398,7 @@ def _on_track_published(
             self._participant_identity is None
             or participant.identity != self._participant_identity
-            or track.source != rtc.TrackSource.SOURCE_MICROPHONE
+            or track.source not in [rtc.TrackSource.SOURCE_MICROPHONE, rtc.TrackSource.SOURCE_SCREENSHARE_AUDIO]
```
same here ^

longcw commented on livekit-agents/livekit/agents/voice/transcription/_utils.py:
```diff
@@ -16,7 +16,7 @@ def find_micro_track_id(room: rtc.Room, identity: str) -> str:
     track_id = None
     for track in p.track_publications.values():
-        if track.source == rtc.TrackSource.SOURCE_MICROPHONE:
+        if track.source in [rtc.TrackSource.SOURCE_MICROPHONE, rtc.TrackSource.SOURCE_SCREENSHARE_AUDIO]:
```
can you add a `audio_sources` arg for `find_micro_track_id` as well?

longcw reviewed (commented):
overall it looks good to me, agree that the option can be `audio_sources` and `video_sources`.

synthobotics commented:
Hi @longcw @theomonnom I am closing this as I opened a different pr, here [https://github.com/livekit/agents/pull/3110](https://github.com/livekit/agents/pull/3110) addressing your comments. Can you please take a look and let me know if you have any questions or concerns
